### PR TITLE
Show all disagreeing annotations

### PR DESCRIPTION
**Pull request by rammba**

### CODE REVIEW

File: src/app/modules/data-set/data-set-instance/data-set-instance.component.html [hunk 1]
```diff
@@ -49,6 +49,15 @@ <h3 class="center-horizontally">Filter: {{instancesType}}</h3>
                         </ng-template>
                     </td>
                 </ng-container>
+                <ng-container matColumnDef="show-annotations" *ngIf="filter == instanceFilter.DisagreeingAnnotations">
+                    <th mat-header-cell *matHeaderCellDef width="5%"> Annotations </th>
+                    <td mat-cell *matCellDef="let instance" width="5%">
+                        <ng-container *ngIf="instance.annotations.length">
+                            <button mat-raised-button color="primary" (click)="showAllAnnotations(instance.annotations, instance.id)">Show</button>
+                        </ng-container>
+                        <ng-container *ngIf="!instance.annotations.length">No annotations</ng-container>
```
Comment (simsimkic) [inline src/app/modules/data-set/data-set-instance/data-set-instance.component.html]: I made a fix commit on the master branch of the _platform_ repository - instances that do not have annotations will no longer appear, so we will no longer need this line of code.

File: src/app/modules/data-set/data-set-instance/data-set-instance.component.ts [hunk 2]
```diff
@@ -15,9 +19,11 @@ import { AnnotationStatus, InstanceType } from '../model/enums/enums.model';
 export class DataSetInstanceComponent implements OnInit {
 
   @Input() public instances: DataSetInstance[] = [];
-  @Input() public instancesType: string = '';
+  @Input() public filter: InstanceFilter = InstanceFilter.All;
+  public instanceFilter = InstanceFilter;
```
Comment (simsimkic) [inline src/app/modules/data-set/data-set-instance/data-set-instance.component.ts:23]: What is the difference between these two filters?

File: src/app/modules/data-set/dialogs/disagreeing-annotations-dialog/disagreeing-annotations-dialog.component.html [hunk 3]
```diff
@@ -0,0 +1,26 @@
+<h2 mat-dialog-title>Disagreeing annotations</h2>
+<mat-dialog-content class="mat-typography">
+    <form>
+        <table cellspacing="2">
+            <ng-container *ngFor="let annotation of annotationsWithInstanceId.annotations">
```
Comment (simsimkic) [inline src/app/modules/data-set/dialogs/disagreeing-annotations-dialog/disagreeing-annotations-dialog.component.html]: _instanceWithDisagreeingAnnotations_ is probably better than _annotationsWithInstanceId_

**Review by simsimkic — CHANGES_REQUESTED** [review]
Few comments to address.

File: src/app/modules/data-set/data-set-instance/data-set-instance.component.ts (hunk 2 above)
Comment (rammba) [reply src/app/modules/data-set/data-set-instance/data-set-instance.component.ts:23]: First one,  _filter_, is enum value used in **.ts** to filter instances.
Secondly, _instanceFilter_, is enum type which is used in **.html** in **ngFor**. Angular throws error if not compared directly with enum type.

**Review by simsimkic — APPROVED** [review]

**Pull request merged**